src/static_analysis/manifest.rs

```
@@ -816,6 +846,38 @@ mod tests {
                    "android.permission.WRITE_EXTERNAL_STORAGE");
         assert!(Permission::from_str("Razican").is_err());
     }
+
+    #[test]
+    fn it_loads_metadata_from_apktool() {
+        let metadata = PackageMetadata::from_apktool("tests/apktool/full_apktool.yml", false)
+            .unwrap();
+
+        assert_eq!(14, metadata.get_min_sdk());
+        assert_eq!(Some(23), metadata.get_target_sdk());
+        assert_eq!(7052, metadata.get_version_number());
+        assert_eq!("0.7.5", metadata.get_version_str());
+    }
+
+    #[test]
+    fn it_loads_metadata_from_apktool_partial() {
+        let metadata = PackageMetadata::from_apktool("tests/apktool/partial_apktool.yml", false)
+            .unwrap();
+
+        assert_eq!(0, metadata.get_min_sdk());
+        assert_eq!(None, metadata.get_target_sdk());
+        assert_eq!(0, metadata.get_version_number());
+        assert_eq!("0.7.5", metadata.get_version_str());
+    }
+
+    #[test]
+    fn it_do_not_load_on_invalid_apktool() {
+        let metadata = PackageMetadata::from_apktool("tests/apktool/invalid.yml", false).unwrap();
```

This should probably return an error.

---

Ok, I will change it

---

Changed at https://github.com/SUPERAndroidAnalyzer/super/pull/94/commits/bef61ad0d027310ba8e0c047714c130991e7be6d